source/core/ib_particle.cc

```diff
@@ -1,5 +1,72 @@
 #include <core/ib_particle.h>
 
 
-extern template class IBParticle<2>;
-extern template class IBParticle<3>;
+template <int dim>
+void
+IBParticle<dim>::initialise_all()
+{
```

Comment:
Why an initialize function and not a constructor?

Reply:
I have multiple initialize function for different parameters. But maybe using 1 constructor and 1 initialize could do the trick also.